core/src/main/java/nu/yona/server/analysis/service/WeekActivityDto.java

```diff
@@ -121,24 +121,29 @@ public static WeekActivityDto createInstanceInactivity(UserAnonymizedDto userAno
 				includeDetail ? Optional.of(0) : Optional.empty(), new HashMap<>(),
 				IntervalActivity.hasPrevious(goal, startOfWeek, ChronoUnit.WEEKS),
 				IntervalActivity.hasNext(startOfWeek, ChronoUnit.WEEKS));
-		weekActivity.createRequiredInactivityDays(userAnonymized,
+		weekActivity.createRequiredInactivityDays(userAnonymized, earliestPossibleDate,
 				userAnonymized.getGoalsForActivityCategory(goal.getActivityCategory()), levelOfDetail, missingInactivities);
 		return weekActivity;
 	}
 
-	public void createRequiredInactivityDays(UserAnonymizedDto userAnonymized, Set<GoalDto> goals, LevelOfDetail levelOfDetail,
-			Set<IntervalInactivityDto> missingInactivities)
+	public void createRequiredInactivityDays(UserAnonymizedDto userAnonymized, LocalDate earliestPossibleDate, Set<GoalDto> goals,
+			LevelOfDetail levelOfDetail, Set<IntervalInactivityDto> missingInactivities)
 	{
 		// if the batch job has already run, skip
 		if (dayActivities.size() == 7)
 		{
 			return;
 		}
+		ZonedDateTime earliestPossibleDateTime = earliestPossibleDate.atStartOfDay(userAnonymized.getTimeZone());
 		// notice this doesn't take care of user time zone changes during the week
 		// so for consistency it is important that the batch script adding inactivity does so
 		for (int i = 0; i < 7; i++)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS135)